src/scenes/challengeMode.js

```diff
@@ -1,4 +1,30 @@
-const challengeMode = function()
-{
-    background(0, 0, 255);
-}
\ No newline at end of file
+import Scene from "../objects/scene.js";
+import ConstantGate from "../objects/constantGate.js";
+import AndGate from "../objects/andGate.js";
+
+/**
+ * This scene will manage the game's challenge mode.
+ */
+export default class ChallengeMode extends Scene {
+    constructor() {
+        super();
+    }
+
+    setup() {
+    	this.constant1;
+    	this.constant2;
+    	this.gate;
+    	let img = loadImage("assets/textures/gates/No Input Buffer.png");
```

Comment:
Is it possible to have each gate we might `new` up here load its own image somehow? It would be nice to be able to call `new AndGate` and it can take care of loading its own image without the dev needing to know which png file to use every time.

Reply:
Resolved via a factory object